recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
+        return self._parasite
+
+    @property
+    def host(self) -> str:
+        return self._host
+    
+    def __eq__(self, other):
+        if type(self) == type(other):
+            return self._parasite == other.parasite and self._host == other.host
+        return False
+    
+    def __hash__(self):
+        return hash((self._parasite, self._host))
+    
+    def __repr__(self):
+        return "%s(%s, %s)" % (type(self).__name__, self._parasite, self._host)
+
+class EventType(Enum):
+    """
+    EventType is the type of event node.
+    """
+    COSPECIATION = 1
+    DUPLICATION = 2
+    TRANSFER = 3
+    LOSS = 4
+    TIPTIP = 5
+
+class Event(ABC):
```

Comment:
`ABC` stands for abstract base class.

Reply:
This means you cannot create an object of type `Event` but has to create an object of class, say, `Loss`.

Reply:
That's cool!  I didn't know that this existed!